1012 fix qicore 2025 bugs tuple element value does not have a resulttypespecifier

baseTwo opened the pull request:
This PR fixes QICore 2025 ELM compilation bugs related to missing `resultTypeSpecifier` properties on tuple element values. The fix enhances the `MissingResultTypeSpecifierCorrector` preprocessor to handle additional cases where type specifiers are missing in tuple operations, particularly in CQL expressions that return complex structured data.

Key changes:
- Enhanced the MissingResultTypeSpecifierCorrector with additional logic to handle tuple element type specifications
- Removed previously skipped CQL files from packager configurations
- Updated submodule references and added generated C# files for newly supported measures

baseTwo committed:
Propagate resultTypeSpecifier from Query -> ReturnClause -> Tuple

baseTwo committed:
Fixes generate additional C#

baseTwo committed:
Add debug logging to MissingResultTypeSpecifierCorrector

Enhanced logging to improve traceability of corrections for missing `resultTypeSpecifier` values in `Query`, `ReturnClause`, and `Tuple` nodes.

- Added logs for `Query` nodes to indicate when `ReturnClause.resultTypeSpecifier` is set to the `elementType` of a `ListTypeSpecifier`.
- Added logs for `ReturnClause` nodes to indicate when `Tuple.resultTypeSpecifier` is set to the `TupleTypeSpecifier`.
- Added logs for `Tuple` nodes to indicate when missing `Tuple.element.value.resultTypeSpecifier` values are set from the `TupleTypeSpecifier`.

These changes provide additional context, such as `locator` and `expressionKey`, to aid debugging.

ewoutkramer commented on Cql/Cql.Compiler/Preprocessing/MissingResultTypeSpecifierCorrector.cs:
```diff
@@ -18,17 +19,68 @@ internal class MissingResultTypeSpecifierCorrector(ILogger<MissingResultTypeSpec
+                {
+                    tupleElement.value.resultTypeSpecifier ??= tupleTypeSpecifier.element[index].elementType;
+                }
+            }
```
Why would copilot have turned the switch into consecutive anonymous code blocks with nested ifs? At least, it looks like that on my diff?

I think these could all be neat cases on a switch, expressing concisely the different types of nodes that are corrected here.

baseTwo replied:
They should not be in a switch, because they could go into more than one clause, eg. an Expression or a Query

baseTwo replied:
I added a code comment to make it more clear what the idea is and not to refactor it to a switch

baseTwo committed:
Clarify ElmTreeWalker pattern matching logic

Added comments to explain the behavior and intent of the
pattern matching logic in the `ElmTreeWalker` creation
within the `MissingResultTypeSpecifierCorrector` class.
The comments highlight the importance of processing order
and why a switch expression is not suitable. These changes
improve code readability and help prevent unintended
modifications that could disrupt functionality.
